### PR TITLE
codewhisperer: enable UTG(Unit test generator) for java/python

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description

* Enable UTG (unit test generator) feature flag for CrossFile userGroup
* Make changes to ensure supplemental context telemetry being sent regardless the userGroup it belongs (we need `metadata.codewhispererSupplementalContextIsUtg` field to differentiate UTG feature acceptance rate vs. CrossFile feature)
## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt [hunk 1]
```diff
@@ -512,31 +512,27 @@ class CodeWhispererService {
         // 2. supplemental context
         val startFetchingTimestamp = System.currentTimeMillis()
         val isTstFile = FileContextProvider.getInstance(project).isTestFile(psiFile)
-        val supplementalContext = if (CodeWhispererUserGroupSettings.getInstance().getUserGroup() == CodeWhispererUserGroup.CrossFile && !isTstFile) {
```
Comment (Will-ShaoHua) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:515]: remove this `if` block

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:515]: delegate this part of logic to here https://github.com/aws/aws-toolkit-jetbrains/pull/3755/files#diff-7aced45233ad4c7e904e77408bc9f4e8545b011324667d8918129994f555b7a8R114

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt [hunk 2]
```diff
@@ -110,7 +112,11 @@ class DefaultCodeWhispererFileContextProvider(private val project: Project) : Fi
         val isTst = isTestFile(psiFile)
 
         val chunks = if (isTst && targetContext.programmingLanguage.isUTGSupported()) {
-            extractSupplementalFileContextForTst(psiFile, targetContext)
+            if (CodeWhispererUserGroupSettings.getInstance().getUserGroup() == CodeWhispererUserGroup.Utg) {
+                extractSupplementalFileContextForTst(psiFile, targetContext)
+            } else {
+                emptyList()
```
Comment (Will-ShaoHua) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt:118]: not returning null for purpose, null is returned only for non-supported language. We will send supplemental context telemetry no matter which user group the user belongs

**Review by andrewyuq — APPROVED** [review]

**Review by vchikoti1998 — APPROVED** [review]

File: .changes/next-release/feature-581591a9-066f-4696-bc5c-1eb8cd01b642.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "feature",
+  "description" : "CodeWhisperer: Improve file context fetching logic"
```
Comment (Will-ShaoHua) [inline .changes/next-release/feature-581591a9-066f-4696-bc5c-1eb8cd01b642.json:3]:
```suggestion
  "description" : "CodeWhisperer: Improve suggestion quality with enhanced file context fetching"
```

**Pull request merged**